### DISCUSSION TITLE
How to integrate 3rd-party OAuth2 provider,like logto,with FastAPI and OpenAPI?

### DISCUSSION THREAD

**Discussion by yixinBC** [fastapi/fastapi#11236] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# I maintain a repo for that question: https://github.com/yixinBC/logto-fastapi-test
from fastapi import FastAPI, Depends
from fastapi.security import OpenIdConnect

app = FastAPI()


oidc = OpenIdConnect(
    openIdConnectUrl="http://localhost:3001/oidc/.well-known/openid-configuration"
)


@app.get("/foo")
async def bar(token=Depends(oidc)):
    return token
```

### Description

I'm trying to use [logto](https://github.com/logto-io/logto) in my FastAPI app.
When I followed logto's [Python SDK guide](https://docs.logto.io/sdk/python/#fetch-access-token-for-the-api-resource),I found the SDK will create a client for authentication.I found it's incompatible with FastAPI's OpenAPI OAuth2 flow.
I tried to integrate logto into a standard OAuth2 flow that can be used for OpenAPI's authentication.I've read the source code of both logto Python SDK and FastAPI's implemetion of OAuth2 in fastapi.security,but just have no idea.
I've also read #12 ,actually I haven't understand the four OAuth 2.0 flows or OpenIdConnect.But I tried to use OpenID and got numerous errors.Perhaps my logto OpenID configuration is incorrect?
I am far from an Oauth2 or backend-development expert, so if I've made any commonsensical mistakes,plz just point them out directly.
Or is it just impossible to include logto(Or other similar products such as Auth0 and casdoor) into OpenAPI's OAuth2 flow?

### Operating System

Windows

### FastAPI Version

0.110.0

### Pydantic Version

2.6.3

### Python Version

Python 3.12.0

**Comment by yixinBC** [comment, marked as answer]
## Update
have read #12 again,and now I think I have enough imformation to solve that

**Comment by jtiagobsouza** [comment]
vcpode usar o /console/api-resources

**Comment by raceychan** [comment]
Yeah, the problem with current implementation of Security in FastAPI is that the OpenAPI schema generation  is coupled with the actual auth header verification(they use the same class).

You might want to subclass the OpenIdConnect class then rewrite its __call__ method with OIDC.
